src/oscar/apps/communication/utils.py

```diff
@@ -156,18 +156,22 @@ def prepare_attachments(self, attachments):
 
         return content_attachments, file_attachments
 
-    def get_base_context(self):
+    def get_base_context(self, **kwargs):
         """
         Return context that is common to all emails
         """
-        return {'site': Site.objects.get_current()}
+        kwargs.update({
+            "site": Site.objects.get_current(kwargs.get("request")),
+        })
+        return kwargs
 
     def get_messages(self, event_code, extra_context=None):
         """
         Return rendered messages
         """
+        if extra_context is None:
+            extra_context = {}
         context = self.get_base_context()
```

Comment:
You forgot to pass the `extra_kwargs`

Reply:
You were quick, haha! But I've changed it :)